File: hw/quina/u2/hw19Inheritance/FarmSystemPy/ec_edu_espe_model/Cow.py
"""
Author: Maryuri Quiña
Description: Farm System – Inheritance example in Python.
"""

from datetime import date
from .FarmAnimal import FarmAnimal
from .Cage import Cage


class Cow(FarmAnimal):
    def __init__(
        self,
        is_producing_milk: bool,
        liters_a_day: float,
        animal_id: int,
        breed: str,
        born_on: date,
        gender: str,
        is_able_to_reproduce: bool,
        weight: float,
        cage: Cage,
    ):
        super().__init__(animal_id, breed, born_on, gender, is_able_to_reproduce, weight, cage)
        self.is_producing_milk = is_producing_milk
        self.liters_a_day = liters_a_day

    def __str__(self) -> str:
        return (
            f"Cow(is_producing_milk={self.is_producing_milk}, "
            f"liters_a_day={self.liters_a_day}, {super().__str__()})"
        )

File: hw/rojas/u2/hw19Inheritance/hw19InheritancePythonAndJavaScript/hw20JavaScript/ec.edu.espe.model/Cow.js
import { FarmAnimal } from './FarmAnimal.js';

export class Cow extends FarmAnimal {
    constructor(id, breed, bornOn, gender, isAbleToReproduce, weight, cage, isProducingMilk, littersADay) {
        super(id, breed, bornOn, gender, isAbleToReproduce, weight, cage);
        this.isProducingMilk = isProducingMilk;
        this.littersADay = littersADay;
    }

    milk() {
        if (this.isProducingMilk) {
            console.log(`Ordeñando... Se obtuvieron ${this.littersADay} litros.`);
            return this.littersADay;
        } else {
            console.log("Esta vaca no está produciendo leche actualmente.");
            return 0;
        }
    }
}